newrelic/resource_newrelic_cloud_gcp_link_account.go

```diff
@@ -0,0 +1,177 @@
+package newrelic
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/newrelic/newrelic-client-go/pkg/cloud"
+)
+
+func resourceNewRelicCloudGcpLinkAccount() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceNewRelicCloudGcpLinkAccountCreate,
+		ReadContext:   resourceNewRelicCloudGcpLinkAccountRead,
+		UpdateContext: resourceNewRelicCloudGcpLinkAccountUpdate,
+		DeleteContext: resourceNewRelicCloudGcpLinkAccountDelete,
+		Schema: map[string]*schema.Schema{
+			"account_id": {
+				Type:        schema.TypeInt,
+				Description: "accountID of newrelic account",
+				Required:    true,
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Description: "name of the linked account",
+				Required:    true,
+			},
+			"project_id": {
+				Type:        schema.TypeString,
+				Description: "project id of the Gcp account",
+				Required:    true,
+			},
+		},
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(10 * time.Second),
+		},
+	}
+}
+
+func resourceNewRelicCloudGcpLinkAccountCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	providerConfig := meta.(*ProviderConfig)
+	client := providerConfig.NewClient
+	accountID := selectAccountID(providerConfig, d)
+	linkAccountInput := expandGcpCloudLinkAccountInput(d)
+	var diags diag.Diagnostics
+	retryErr := resource.RetryContext(ctx, d.Timeout(schema.TimeoutCreate), func() *resource.RetryError {
+		cloudLinkAccountPayload, err := client.Cloud.CloudLinkAccountWithContext(ctx, accountID, linkAccountInput)
+		if err != nil {
+			return resource.NonRetryableError(err)
+		}
+
+		if len(cloudLinkAccountPayload.Errors) > 0 {
+			for _, err := range cloudLinkAccountPayload.Errors {
+				if strings.Contains(err.Message, "The ARN you entered does not permit the correct access to your AWS account") {
+					return resource.RetryableError(fmt.Errorf("%s : %s", err.Type, err.Message))
+				}
+				diags = append(diags, diag.Diagnostic{
```

Comment:
The retry mechanism could add the same error multiple times here. I'm not 100% sure this will happen, but want to double check.